api/channel.go

```diff
@@ -843,17 +843,25 @@ func setLastViewedAt(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	data := model.StringInterfaceFromJson(r.Body)
 	newLastViewedAt := int64(data["last_viewed_at"].(float64))
+	channelName := data["channel_name"].(string)
 
 	Srv.Store.Channel().SetLastViewedAt(id, c.Session.UserId, newLastViewedAt)
 
-	preference := model.Preference{
+	chanPref := model.Preference{
 		UserId:   c.Session.UserId,
-		Category: model.PREFERENCE_CATEGORY_LAST,
+		Category: c.TeamId,
```

Comment:
Never mind, I saw your code below to delete the user's team-specific preferences when they leave a team. I actually like that much better. We don't need to change the existing themes code, but all future team-specific preferences should follow that pattern in the future